WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/Carriers and Rates/ShippingLabelCarrierRow.swift

```diff
@@ -0,0 +1,67 @@
+import SwiftUI
+
+struct ShippingLabelCarrierRow: View, Identifiable {
+
+    let id = UUID()
+    let title: String
+    let subtitle: String
+    let price: String
+    let image: UIImage?
+
+    var body: some View {
+        HStack(spacing: Constants.hStackSpacing) {
+            if let image = image {
+                ZStack {
+                    Image(uiImage: image).frame(width: Constants.imageSize, height: Constants.imageSize)
+                }
+                .frame(width: Constants.zStackWidth)
+            }
+            VStack(alignment: .leading,
+                   spacing: Constants.vStackSpacing) {
+                HStack {
+                    Text(title)
+                        .bodyStyle()
+                    Spacer()
+                    Text(price)
+                        .bodyStyle()
+                }
+                Text(subtitle)
+                    .footnoteStyle()
+            }
+        }
+        .padding([.top, .bottom], Constants.hStackPadding)
+        .padding([.leading, .trailing], Constants.padding)
+        .frame(minHeight: Constants.height)
+        .contentShape(Rectangle())
+    }
+}
+
+private extension ShippingLabelCarrierRow {
+    enum Constants {
+        static let zStackWidth: CGFloat = 48
+        static let vStackSpacing: CGFloat = 8
+        static let hStackSpacing: CGFloat = 25
+        static let hStackPadding: CGFloat = 10
+        static let height: CGFloat = 60
+        static let imageSize: CGFloat = 40
+        static let padding: CGFloat = 16
+    }
+}
+
+struct ShippingLabelCarrierRow_Previews: PreviewProvider {
+    static var previews: some View {
+        ShippingLabelCarrierRow(title: "UPS Ground",
+                                subtitle: "3 business days",
+                                price: "$2.49",
+                                image: UIImage(named: "shipping-label-ups-logo")!)
+            .previewLayout(.fixed(width: 375, height: 60))
+            .previewDisplayName("With Image")
+
+        ShippingLabelCarrierRow(title: "UPS Ground",
+                                subtitle: "3 business days",
+                                price: "$2.49",
+                                image: nil)
+            .previewLayout(.fixed(width: 375, height: 60))
+            .previewDisplayName("Without Image")
+    }
```

Comment:
Do you think it's useful to have a large font size preview as well? 

```suggestion
        ShippingLabelCarrierRow(title: "UPS Ground",
                                subtitle: "3 business days",
                                price: "$2.49",
                                image: nil)
            .frame(maxWidth: 375, minHeight: 60)
            .previewLayout(.sizeThatFits)
            .environment(\.sizeCategory, .accessibilityExtraExtraExtraLarge)
            .previewDisplayName("Large Font Size Roar!")

    }
```

<img src="https://user-images.githubusercontent.com/198826/119004275-a52fef80-b94b-11eb-9a1f-5455722f0867.png" width="400">